### PR TITLE
Feature: Tests generator tool

### PR DESCRIPTION
**Pull request by DonatasBrazionis**

### CODE REVIEW

**Review by MartynasZilinskas — APPROVED** [review]

File: tools/test-generator-cli/src/cli-arguments.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * Handles all CLI commands and arguments.
+ */
+export const CLIHandler = yargs
+    .showHelpOnFail(true)
+    .help("h", "Show help")
+    .alias("h", "help")
+    .version(`Current version: ${GetVersion()}`)
+    .alias("v", "version")
+    // CLI options
+    .option("path", {
```
Comment (DeividasBakanas) [inline tools/test-generator-cli/src/cli-arguments.ts]: Consider renaming to "project".

File: tools/test-generator-cli/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            Logger.Info(`Test file was changed in "${testsCasesPath}/${fileName}".`);
+            const startBuild = Logger.Info(`Generating tests for "${testsCasesPath}"...`);
+            await TestsGenerator(testsCasesPath);
+            Logger.Debug(`Generated tests after ${(Date.now() - startBuild)}ms`);
+        }
+    });
+}
+
+(async (argumentsObject: CLIArgumentsObject) => {
+    let cwd: string;
```
Comment (DeividasBakanas) [inline tools/test-generator-cli/src/index.ts]: Can be replaced with:
```ts
const cwd = argumentObject.path || process.cwd();
```

File: tools/test-generator-cli/src/tests-generator.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+export async function TestsGenerator(testsCasesPath: string): Promise<void> {
+    const defaultTemplatePath = FixSep(path.join(testsCasesPath, "..", DEFAULT_TEMPLATE_FILE_NAME));
+    if (!await fs.pathExists(defaultTemplatePath)) {
+        Logger.Error(`Not found package default template. Expected file location: "${defaultTemplatePath}"`);
+        return;
+    }
+
+    const casesDirPaths = await fastGlob([
```
Comment (DeividasBakanas) [inline tools/test-generator-cli/src/tests-generator.ts]: Reconsider formatting:

```ts
const casesDirPaths = await fastGlob(
    [
        `${testsCasesPath}/*`
    ],
    {
        onlyDirs: true,
        ignore: [`**/${GENERATED_TESTS_DIR_NAME}/**`]
    }
);
```

or 

```ts
const casesDirPaths = await fastGlob(
    [`${testsCasesPath}/*`],
    {
        onlyDirs: true,
        ignore: [`**/${GENERATED_TESTS_DIR_NAME}/**`]
    }
);
```

File: tools/test-generator-cli/src/tests-helpers.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import * as path from "path";
+import * as handlebars from "handlebars";
+
+const TAB_STRING = "    ";
+
+export function FixSep(pathname: string): string {
+    return pathname.split(path.sep).join("/");
+}
+
+export function Tab(size: number = 1): string {
```
Comment (DeividasBakanas) [inline tools/test-generator-cli/src/tests-helpers.ts]: Is this helper still useful?

**Review by DeividasBakanas — CHANGES_REQUESTED** [review]

**Review by DeividasBakanas — APPROVED** [review]

**Pull request merged**